pkg/controller/scyllacluster/resource.go

```diff
@@ -785,24 +830,29 @@ func containerPorts(c *scyllav1.ScyllaCluster) []corev1.ContainerPort {
 			Name:          "node-exporter",
 			ContainerPort: 9100,
 		},
+		{
+			Name:          "thrift",
+			ContainerPort: 9160,
+		},
 	}
 
-	if c.Spec.Alternator.Enabled() {
+	if c.Spec.Alternator != nil {
 		ports = append(ports, corev1.ContainerPort{
-			Name:          "alternator",
-			ContainerPort: c.Spec.Alternator.Port,
-		})
-	} else {
-		ports = append(ports, corev1.ContainerPort{
-			Name:          "cql",
-			ContainerPort: 9042,
-		}, corev1.ContainerPort{
-			Name:          "cql-ssl",
-			ContainerPort: 9142,
-		}, corev1.ContainerPort{
-			Name:          "thrift",
-			ContainerPort: 9160,
+			Name:          alternatorTLSPortName,
+			ContainerPort: alternatorTLSPort,
 		})
+
+		enableHTTP := c.Spec.Alternator.InsecureEnableHTTP
+		if c.Spec.Alternator.Port != 0 || (enableHTTP != nil && *enableHTTP) {
+			insecurePort := int32(alternatorInsecurePort)
+			if c.Spec.Alternator.Port != 0 {
+				insecurePort = c.Spec.Alternator.Port
+			}
+			ports = append(ports, corev1.ContainerPort{
+				Name:          alternatorInsecurePortName,
+				ContainerPort: insecurePort,
+			})
+		}
```

Comment:
This is copy pasted from servicePorts, please extract common method of getting alternator ports and append them.